src/main/scala/cromwell/util/GoogleCloudStoragePath.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.util
+
+import java.net.URI
+
+object GoogleCloudStoragePath {
+  def apply(value: String): GoogleCloudStoragePath = {
+    val gsUriRegex = """gs://([^/]*)/(.*)""".r
+    value match {
+      case gsUriRegex(bucket, objectName) => GoogleCloudStoragePath(bucket, objectName)
+      // TODO: error handle:
+      case _ => throw new IllegalArgumentException()
+    }
+  }
+
+  def formUriString(bucket: String, objectName: String) = "gs://" + bucket + "/" + objectName
```

Comment:
I'd make this just `uriString`, scala prefers removing stuff like `get` the like (or `form` in this case) for things which are pure functions like this